devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,238 @@
+        }
+
+        main "alarm"
+        details(["alarm", "off", "tamper"])
+    }
+}
+
+def installed() {
+    initialize()
+    def numberOfSounds = 8
+    state.soundTable = []
+    for (def i in 0..numberOfSounds) {
+        state.soundTable[i] = 0
+    }
+    sendEvent(name: "alarm", value: "off", isStateChange: true)
+}
+
+def updated() {
+    initialize()
+}
+
+def initialize() {
+    if (!childDevices) {
+        addChildren()
+    }
+    sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+}
+
+def parse(String description) {
+    def result = null
+    def cmd = zwave.parse(description)
+    if (cmd) {
+        result = zwaveEvent(cmd)
+    }
+    log.debug "Parse returned: ${result.inspect()}"
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+    createEvents(cmd.value)
+}
+
+def createEvents(value) {
+    sendEvent(name: "alarm", value: value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+    def encapsulatedCommand = cmd.encapsulatedCommand()
+    def endpoint = cmd.sourceEndPoint
+    if (endpoint == 1) {
+        zwaveEvent(encapsulatedCommand)
+    } else {
+        String childDni = "${device.deviceNetworkId}:$endpoint"
+        def child = childDevices.find { it.deviceNetworkId == childDni }
+
+        child?.handleZWave(encapsulatedCommand)
+    }
+}
+
+def on() {
+    setActiveSound(1)
+    def cmds = []
+    cmds <<encap(zwave.basicV1.basicSet(value: 0xFF).format())
+    cmds
+}
+
+def off() {
+    setActiveSound(1)
+    def cmds = []
+    cmds <<encap(zwave.basicV1.basicSet(value: 0x00).format())
+    cmds
+}
+
+def strobe() {
+    on()
+}
+
+def siren() {
+    on()
+}
+
+def both() {
+    on()
+}
+
+def ping() {
+    encap(zwave.basicV1.basicGet())
+}
+
+private addChildren() {
+    def numberOfSounds = 8
+    for (def endpoint : 2..numberOfSounds) {
+        try {
+            String childDni = "${device.deviceNetworkId}:$endpoint"
+            def componentLabel = device.displayName[0..-2] + "$endpoint"
+
+            addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+                    completedSetup: true,
+                    label         : componentLabel,
+                    isComponent   : true,
+                    componentName : "Aeotec Doorbell 6 $endpoint",
+                    componentLabel: "Aeotec Doorbell 6 $endpoint"
+            ])
+        } catch (Exception e) {
+            log.debug "Excep: ${e} "
+        }
+    }
+}
+
+private encap(cmd, endpoint = null) {
+    if (endpoint) {
+        zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+    } else {
+        cmd
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
```

Comment:
will never be called with an endpoint argument, because you're attempting to pass any multichannel-encapsulated command to the child device.